### PR TITLE
feat: Issue113

### PR DESCRIPTION
**Pull request by JulienBarlet**
## Type of modification

* [ ] Breaking change
* [x] New Feature
* [ ] Bug Fix
* [ ] Chore (refactor, documentation, tests... all the changes with no impact on ARA functionalities.) 

## Changes description

When ARA provide error list, for example in the error page, the matching errors list will be sorted by recent failed executions by default 

## Technical description

On Error repository, set the sorting direction with DESC instead of ASC

## PR CheckList

Please make sure your PullRequest respect all those items :

* [x] Your PR's title has the prefix : `feat:`, `fix:` or `chore:`
* [x] Unless your PR is related to a Hotfix (and approved by a ARA maintener), the targeted branch for it is the branch `version/X.Y.Z` and **not** `master`
* [x] You have asked a review from one of the ARA maintainer in your PR.
* [x] If your PR is related to an issue, add the issue's number in it.
* [x] All the code you added is documented.
* [x] All the code you added is tested and the  tests are in success.
* [x] You already signed the [Contributor License Agreement](https://github.com/Decathlon/ara/blob/master/contributor-licence-agreement.adoc) and give us the document

### CODE REVIEW

File: database/src/main/java/com/decathlon/ara/repository/custom/impl/ErrorRepositoryImpl.java [hunk 1]
```diff
@@ -56,7 +56,7 @@
 @Slf4j
 public class ErrorRepositoryImpl implements ErrorRepositoryCustom {
 
-    private static final Sort ERROR_SORTING = new Sort(Sort.Direction.ASC,
+    private static final Sort ERROR_SORTING = new Sort(Sort.Direction.DESC,
```
Comment (dktsni) [inline database/src/main/java/com/decathlon/ara/repository/custom/impl/ErrorRepositoryImpl.java]: I've ran your branch locally and still have the errors displayed in Ascending order.
Screenshot 2019-10-21 at 16 13 55
Screenshot 2019-10-21 at 16 14 27


Could you please check if you reproduce the old behavior on your side ?

File: database/src/main/java/com/decathlon/ara/repository/ErrorRepository.java [hunk 2]
```diff
@@ -39,6 +39,8 @@
     // NO projectId: patterns is already restrained to the correct project
     Page<Error> findDistinctByProblemPatternsInOrderById(List<ProblemPattern> patterns, Pageable pageable);
 
+    Page<Error> findDistinctByProblemPatternsInOrderByExecutedScenarioRunExecutionTestDateTimeDesc(List<ProblemPattern> patterns, Pageable pageable);
```
Comment (dktsni) [inline database/src/main/java/com/decathlon/ara/repository/ErrorRepository.java]: Missing documentation

File: server/src/main/java/com/decathlon/ara/service/ProblemService.java [hunk 3]
```diff
@@ -525,19 +525,20 @@ private ProblemWithAggregateDTO toProblemWithAggregate(Problem entity) {
...
-     * @param id        the id of the entity
+     * @param problemId        the id of the entity
      * @param pageable  the meta-data of the requested page
      * @return a page of the errors associated to the problem
      * @throws NotFoundException if the entity id does not exist
      */
     @Transactional(readOnly = true)
-    public Page<ErrorWithExecutedScenarioAndRunAndExecutionDTO> getProblemErrors(long projectId, long id, Pageable pageable) throws NotFoundException {
-        Problem problem = problemRepository.findByProjectIdAndId(projectId, id);
+    public Page<ErrorWithExecutedScenarioAndRunAndExecutionDTO> getProblemErrors(long projectId, long problemId, Pageable pageable) throws NotFoundException {
```
Comment (dktsni) [inline server/src/main/java/com/decathlon/ara/service/ProblemService.java]: Does this new behavior covered by a Unit test ?

Comment (omar-chahbouni-decathlon) [reply server/src/main/java/com/decathlon/ara/service/ProblemService.java]: Done, but honestly I'm not a huge fan of commenting the source code: it should be self-explanatory.
The problem with comments is that when code changes they don't always follow. That is why it is important to work with a code that is easily understood by everyone working with it.

> "_A code never lies, comments sometimes do_" - Ron Jeffries



We can discuss it tomorrow if you want to ;)

Comment (omar-chahbouni-decathlon) [reply server/src/main/java/com/decathlon/ara/service/ProblemService.java]: done

File: database/src/main/java/com/decathlon/ara/repository/ErrorRepository.java (hunk 2 above)
Comment (omar-chahbouni-decathlon) [reply database/src/main/java/com/decathlon/ara/repository/ErrorRepository.java]: done (cf my comment)

Comment (dktsni) [reply database/src/main/java/com/decathlon/ara/repository/ErrorRepository.java]: Here we're not talking about comments but about documentation 😄 
I got your point, but that's also what reviews are supposed to do, make sure that all the changes are made. 
If the code changes and the documentation doesn't in a PR, we can view it and change it :) 

Here is an example : in your method, what happens if one of the param is null ? And the return is always an object or can't be null ?
Also your method (thanks to SpringJDBC...) has 82 characters in it. 82chars of camelCase can be harsh to read to know a purpose of it, spaces can helps a lot in comprehension 😄

**Pull request merged**